### PR TITLE
Feature/add card functionality

### PR DESCRIPTION
**Pull request by adriankulinski**

### CODE REVIEW

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardLoggedValidator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		return clazz.equals(CardLoggedDTO.class);
+	}
+
+	@Override
+	public void validate(Object target, Errors errors) {
+		CardLoggedDTO cardLoggedDTO = (CardLoggedDTO) target;
+		validateActivate(errors, cardLoggedDTO);
+	}
+	
+	private void validateActivate(Errors errors, CardLoggedDTO cardLoggedDTO){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardLoggedValidator.java]: please change name to validateUsersActiveCards

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardNotLoggedValidator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		validateActivate(errors, cardNotLoggedDTO);
+	}
+	
+	private void validateUser(Errors errors, CardNotLoggedDTO cardNotLoggedDTO){
+		if(!userService.getUserByEmailAndNameAndSurname(cardNotLoggedDTO.getEmail(), cardNotLoggedDTO.getName(), cardNotLoggedDTO.getSurname()).isPresent()){
+			errors.reject("card.user_not_exist");
+		}
+	}
+	
+	private void validateActivate(Errors errors, CardNotLoggedDTO cardNotLoggedDTO){
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardNotLoggedValidator.java]: repeated code - please get rid of that (with inheritance maybe? ;) )

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+	
+	private String number;
+	
+	@ManyToOne
+	@JoinColumn(name = "user_id")
+	private User user;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java:24]: Is this field used anywhere? If not please remove it from here.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	
+	@Column(name = "active", columnDefinition = "TINYINT", length = 1)
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java]: There is bit type in sql, please remove columnDefinition and length properties and update table in db

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Points.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+	
+	@Column(name = "points_amount")
+	private int pointsAmount;
+	
+	@ManyToOne
+	@JoinColumn(name = "user_profile_id")
+	private UserProfile userProfile;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Points.java]: Is this field used anywhere? If not please remove it from here.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardLoggedDTO.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.pgs.soft.dto;
+
+public class CardLoggedDTO {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardLoggedDTO.java]: Please change name to AddCardLoggedRequestDTO

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package com.pgs.soft.dto;
+
+public class CardNotLoggedDTO {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java]: Please extend from AddCardLoggedRequestDTO and change name to AddCardRequestDTO

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+	private String number;
+	
+	private String email;
+	
+	private String name; 
+	
+	private String surname;
+	
+	private boolean active;
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java]: remove that flag from DTO

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CardService.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import com.pgs.soft.domain.Card;
+import com.pgs.soft.dto.CardLoggedDTO;
+import com.pgs.soft.dto.CardNotLoggedDTO;
+
+@Service
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CardService.java]: Annotation only in implementation. Please remember that.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+public class CardServiceImpl implements CardService {
+
+	@Autowired
+	private CardRepository cardRepository;
+
+	@Autowired
+	private UserRepository userRepository;
+
+	@Override
+	public void saveForLogged(CardLoggedDTO cardLoggedDTO) {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java]: Please extract repeated code

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	
+	@Override
+	public Set<Card> getCardsByUserId(Integer userId){
+		return cardRepository.findByUserId(userId);
+	}
+	
+	@Override
+	public boolean hasActiveCard(Integer userId){
+		return getCardsByUserId(userId).stream().anyMatch((c) -> c.isActive());
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java:56]: Nice :+1:

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java (hunk 3 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java:24]: Do you mean user? it is used

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java (hunk 4 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java (hunk 8 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CardService.java (hunk 9 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/CardService.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java (hunk 4 above)
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java]: still is tinyint

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java (hunk 11 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java:56]: thanks :D

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardLoggedValidator.java (hunk 1 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardLoggedValidator.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardNotLoggedValidator.java (hunk 2 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/CardNotLoggedValidator.java]: It is ok, we've talked about it

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java (hunk 4 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Card.java]: It is done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardLoggedDTO.java (hunk 6 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardLoggedDTO.java]: AddCardRequestDTO now.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java (hunk 7 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/dto/CardNotLoggedDTO.java]: It is AddCardNotLoggedRequestDTO extends AddCardRequestDTO

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java (hunk 10 above)
Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/CardServiceImpl.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Points.java (hunk 5 above)
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Points.java]: What about that?

Comment (adriankulinski) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/domain/Points.java]: Changed, please take a look

**Pull request merged**